tests/test_boolean.py

```diff
@@ -324,217 +195,419 @@ def test_parse_side_by_side_symbols_with_parens_raise_exception(self):
-        self.assertEqual((~(a & a & a)).simplify(), (~(a & a & a)).simplify())
-        self.assertEqual(a, algebra.parse('~~a', simplify=True))
-
-    def test_cancel(self):
+
+        assert ~a == ~a
+        assert ~a == algebra.NOT(a)
+
+    def test_simplify(self):
         algebra = BooleanAlgebra()
+
         a = algebra.Symbol('a')
-        self.assertEqual(~a, (~a).cancel())
-        self.assertEqual(a, algebra.parse('~~a').cancel())
-        self.assertEqual(~a, algebra.parse('~~~a').cancel())
-        self.assertEqual(a, algebra.parse('~~~~a').cancel())
 
-    def test_demorgan(self):
+        assert a == a.simplify()
+        assert a == (~~a).simplify()
+        assert a == (~~ ~~a).simplify()
+
+        assert ~a == (~a).simplify()
+        assert ~a == (~ ~~a).simplify()
+        assert ~a == (~ ~~ ~~a).simplify()
+
+        assert (~(a & a & a)).simplify() == (~(a & a & a)).simplify()
+        assert (~(a | a | a)).simplify() == (~(a | a | a)).simplify()
+
+    def test_cancel_0(self):
+        """
+        Test .cancel() on python variables
+        """
         algebra = BooleanAlgebra()
+
         a = algebra.Symbol('a')
-        b = algebra.Symbol('b')
-        self.assertEqual(algebra.parse('~(a&b)').demorgan(), ~a | ~b)
-        self.assertEqual(algebra.parse('~(a|b|c)').demorgan(), algebra.parse('~a&~b&~c'))
-        self.assertEqual(algebra.parse('~(~a&b)').demorgan(), a | ~b)
 
+        assert a == (~~a).cancel()
+        assert a == (~~ ~~a).cancel()
+
+        assert ~a == (~a).cancel()
+        assert ~a == (~ ~~a).cancel()
+        assert ~a == (~ ~~ ~~a).cancel()
+
+    def test_cancel_1(self):
+        """
+        Test .cancel() on .parse() results
+        """
+        parse = BooleanAlgebra().parse
+
+        assert parse('a') == parse('~~a').cancel()
+        assert parse('a') == parse('~~ ~~a').cancel()
+
+        assert parse('~a') == parse('~a').cancel()
+        assert parse('~a') == parse('~ ~~a').cancel()
+        assert parse('~a') == parse('~ ~~ ~~a').cancel()
+
+    def test_cancel_2(self):
+        """
+        Test .cancel() on both Python variables and .parse() results
+        """
+        algebra = BooleanAlgebra()
+
+        a, parse = algebra.Symbol('a'), algebra.parse
+
+        assert a == parse('~~a').cancel()
+        assert a == parse('~~ ~~a').cancel()
+
+        assert ~a == parse('~a').cancel()
+        assert ~a == parse('~ ~~a').cancel()
+        assert ~a == parse('~ ~~ ~~a').cancel()
+
+    def test_demorgan(self):
+        parse = BooleanAlgebra().parse
+
+        assert parse('~(a & a)').demorgan() == parse('~a | ~a')
+
+        assert parse('~(a & b)').demorgan() == parse('~a | ~b')
+        assert parse('~(a & b & c)').demorgan() == parse('~a | ~b | ~c')
+
+        assert parse('~(~a & b)').demorgan() == parse('a | ~b')
+        assert parse('~(a & ~b)').demorgan() == parse('~a | b')
+
+    # TODO: enforced order is obscure, must explain
```

Comment:
I meant the lines `585` to `595` below that use the `<` and `>` comparison operators. Is that what you meant?